libdnf/system/state.hpp

```diff
@@ -165,6 +169,22 @@ class State {
     /// @since 5.0
     std::set<std::string> get_package_groups(const std::string & name);
 
+    /// @return The the state for environmental group
+    /// @param id The environmental group id to get the state for.
+    /// @since 5.0
+    EnvironmentState get_environment_state(const std::string & id);
+
+    /// Sets the state for an environmental group id.
+    /// @param id The environmental group id to set the state for.
+    /// @since 5.0
+    void set_environment_state(const std::string & id, const EnvironmentState & environment_state);
```

Comment:
May be I will need an update what is the purpose or mechanism of EnvironmentState.